Wrong BEAT_PATH for metricbeat based community beats

See [discuss](https://discuss.elastic.co/t/make-package-broken-for-community-beats/70511) for background information. @ruflin advised to create an issue for it

DST_FOLDER is not correctly set in [build.sh](https://github.com/elastic/beats/blob/master/dev-tools/packer/docker/xgo-image/base/build.sh) when building a metricbeat based community beat.

"make package" fails as follows:

```
Working directory=/go/src/github.com
Copying main source folder /source to folder /go/src/github.com
Fetching dependencies...
Execute /scripts/before_build.sh github.com/someuser
/scripts/before_build.sh: 13: cd: can't cd to /go/src/github.com/someuser
make[1]: *** [prepare-package-cgo] Error 2
make: *** [package] Error 2
```

The issue seems to be that the argument passed to build.sh is the value of the BEAT_DIR variable in the generated Makefile that doesn't contain the beat name.

Extract from generated Makefile
```
BEATNAME=somebeat
BEAT_DIR=github.com/someuser
```

build.sh is extracting the directory name using the dirname command, thus removing the last part of the BEAT_DIR variable

Extract from build.sh

```
BEAT_PATH=$1
DST_FOLDER=`dirname $GOPATH/src/$BEAT_PATH`
GIT_REPO=$BEAT_PATH
```

In the example BEAT_PATH is "github.com/someuser", DST_FOLDER will be "/go/src/github.com". build.sh creates the DST_FOLDER (/go/src/github.com). Then [before_build.sh](https://github.com/elastic/beats/blob/master/dev-tools/packer/xgo-scripts/before_build.sh) is called with $BEAT_PATH (github.com/someuser) as argument and tries to change directory to /go/src/${1} (/go/src/github.com/someuser) and fails.

To workaround the issue I modified build.sh to:

```
BEAT_NAME=$PACK
BEAT_PATH=$1/$BEAT_NAME
DST_FOLDER=`dirname $GOPATH/src/$BEAT_PATH`
GIT_REPO=$BEAT_PATH
```

There are multiple ways in which this could be solved, I tried a few, but this is the only one I found that didn't have side-effects. Not sure if this is the correct way though.

I'll submit a PR with the changes described above.

Tested on 5b973a073a02ab3753a0b1e61c7947df48f656b0
OS: Ubuntu 14.04
Steps to reproduce:

```
$ cd $GOPATH/src/github.com/someuser
$ cookiecutter $GOPATH/src/github.com/elastic/beats/generate/metricbeat/metricset
project_name [Examplebeat]: somebeat
github_name [your-github-name]: someuser
beat [somebeat]: somebeat
beat_path [github.com/someuser]: github.com/someuser
full_name [Firstname Lastname]: Some User

$ cd somebeat/
$ make setup
$ make update
$ make package
```

--- a/generate/metricbeat/metricset/{{cookiecutter.beat}}/Makefile
+++ b/generate/metricbeat/metricset/{{cookiecutter.beat}}/Makefile
@@ -1,5 +1,5 @@
 BEATNAME={{cookiecutter.beat}}
-BEAT_DIR={{cookiecutter.beat_path}}
+BEAT_DIR={{cookiecutter.beat_path}}/{{cookiecutter.beat}}
 SYSTEM_TESTS=false
 TEST_ENVIRONMENT=false
 ES_BEATS?=./vendor/github.com/elastic/beats

--- a/metricbeat/Makefile
+++ b/metricbeat/Makefile
@@ -51,7 +51,7 @@ configs: python-env
 .PHONY: imports
 imports:
 	mkdir -p include
-	python ${ES_BEATS}/metricbeat/scripts/generate_imports.py ${BEAT_DIR} ${BEATNAME} > include/list.go
+	python ${ES_BEATS}/metricbeat/scripts/generate_imports.py ${BEAT_DIR} > include/list.go
 
 # This is called by the beats packer before building starts
 .PHONY: before-build

--- a/metricbeat/scripts/generate_imports.py
+++ b/metricbeat/scripts/generate_imports.py
@@ -47,9 +47,5 @@ if __name__ == "__main__":
     # First argument is the beat path under GOPATH.
     # (e.g. github.com/elastic/beats/metricbeat)
     go_beat_path = sys.argv[1]
-    beat_name = sys.argv[2]
-
-    if go_beat_path != "github.com/elastic/beats/metricbeat":
-        go_beat_path += "/" + beat_name
 
     generate(go_beat_path)

